File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/parser/style/StyleRulesParamsBuilder.java
package com.github.k1rakishou.chan.core.site.parser.style;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.github.k1rakishou.chan.core.site.parser.PostParser;
import com.github.k1rakishou.core_parser.comment.HtmlTag;
import com.github.k1rakishou.model.data.post.ChanPostBuilder;

import java.util.Objects;

public class StyleRulesParamsBuilder {
    @NonNull
    private CharSequence text;
    @NonNull
    private HtmlTag htmlTag;
    @Nullable
    private PostParser.Callback callback = null;
    @Nullable
    private ChanPostBuilder post = null;

    public StyleRulesParamsBuilder withCallback(PostParser.Callback callback) {
        this.callback = callback;
        return this;
    }

    public StyleRulesParamsBuilder withPostBuilder(ChanPostBuilder post) {
        this.post = post;
        return this;
    }

    public StyleRulesParamsBuilder withText(CharSequence text) {
        this.text = text;
        return this;
    }

    public StyleRulesParamsBuilder withHtmlTag(HtmlTag htmlTag) {
        this.htmlTag = htmlTag;
        return this;
    }

    public StyleRulesParams build() {
        Objects.requireNonNull(text, "text must not bel null");
        Objects.requireNonNull(htmlTag, "htmlTag must not bel null");

        return new StyleRulesParams(text, htmlTag, callback, post);
    }
}

File: Kuroba/app/src/main/java/com/github/k1rakishou/chan/core/site/parser/style/StyleRulesParamsBuilder.kt
package com.github.k1rakishou.chan.core.site.parser.style

import com.github.k1rakishou.chan.core.site.parser.PostParser
import com.github.k1rakishou.core_parser.comment.HtmlTag
import com.github.k1rakishou.model.data.post.ChanPostBuilder
import java.util.Objects

class StyleRulesParamsBuilder {
  private var text: CharSequence? = null
  private var htmlTag: HtmlTag? = null
  private var callback: PostParser.Callback? = null
  private var post: ChanPostBuilder? = null
  private var forceHttpsScheme = true

  fun withCallback(callback: PostParser.Callback?): StyleRulesParamsBuilder {
    this.callback = callback
    return this
  }

  fun withPostBuilder(post: ChanPostBuilder?): StyleRulesParamsBuilder {
    this.post = post
    return this
  }

  fun withText(text: CharSequence?): StyleRulesParamsBuilder {
    this.text = text
    return this
  }

  fun withHtmlTag(htmlTag: HtmlTag?): StyleRulesParamsBuilder {
    this.htmlTag = htmlTag
    return this
  }

  fun forceHttpsScheme(forceHttpsScheme: Boolean): StyleRulesParamsBuilder {
    this.forceHttpsScheme = forceHttpsScheme
    return this
  }

  fun build(): StyleRulesParams {
    Objects.requireNonNull<CharSequence?>(text, "text must not bel null")
    Objects.requireNonNull<HtmlTag?>(htmlTag, "htmlTag must not bel null")

    return StyleRulesParams(text!!, htmlTag!!, callback, post, forceHttpsScheme)
  }
}